import/webview/WebView.qml

```diff
@@ -31,19 +31,32 @@ RawWebView {
         webview.loadFrameScript("chrome://embedlite/content/embedhelper.js");
         webview.loadFrameScript("chrome://embedlite/content/SelectAsyncHelper.js");
         webview.addMessageListeners([
-            "embed:linkclicked",
-            "embed:filepicker",
-            "embed:selectasync",
-            "embed:alert",
-            "embed:confirm",
-            "embed:prompt",
-            "embed:auth",
-            "embed:permissions",
-            "embed:login",
-            "Content:ContextMenu"]);
+                                        "embed:linkclicked",
+                                    ]);
+    }
+
+    PickerOpener {
+        id: pickerOpener
+
+        property QtObject pageStackOwner: helper.findParentWithProperty(webView, "pageStack")
+
+        pageStack: pageStackOwner ? pageStackOwner.pageStack : undefined
+        contentItem: webview
+    }
+
+    PopupOpener {
+        id: popupOpener
+
+        pageStack: pickerOpener.pageStack
+        parentItem: webview
+        contentItem: webview
     }
 
     onRecvAsyncMessage: {
+        if (pickerOpener.handlesMessage(message) || popupOpener.handlesMessage(message)) {
+            return
+        }
+
         // cache some symbol resolutions in var properties in the function closure
         var privData = helper
```

Comment:
Will add a commit on top that cleans up this and Timer parts. This I missed but originally didn't want to tamper ```helper``` now as this is growing fast.